compiler/passes/checkParsed.cpp

```diff
@@ -359,7 +359,11 @@ checkFunction(FnSymbol* fn) {
     USR_FATAL_CONT(fn, "method 'these' must have parentheses");
 
   if (fn->thisTag != INTENT_BLANK && fn->isMethod() == false) {
-    USR_FATAL_CONT(fn, "'this' intents can only be applied to methods");
+    if (fn->thisTag == INTENT_TYPE) {
+      USR_FATAL_CONT(fn, "Missing type for type method");
```

Comment:
I feel like this could use the word secondary to try to make it clearer where the type is missing: "Missing type for secondary type method".